test/builder-api/test.sh

```diff
@@ -1,34 +1,283 @@
 #!/bin/bash
 
-if [ "$I_MEAN_IT" != "1" ]; then 
-  echo "**** THESE TESTS WILL DESTROY YOUR DATABASES ****"
-  echo "**** AND I MEAN THE REALLY REAL ONES ****"
-  echo
-  echo "Set I_MEAN_IT=1 to continue"
-  exit 500
+# This script exists to run our integration tests.
+
+set -e
+
+# Check to see if a command exists
+exists() {
+  if command -v $1 >/dev/null 2>&1
+  then
+    return 0
+  else
+    return 1
+  fi
+}
+
+# base_dir is the root of the habitat project.
+base_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
+tmp_dir=/tmp
+pg_tmp_version=2.3
+export BLDR_FULL_TEST_RUN=1
+unset npm_config_prefix
+pg_svc_dir="/hab/svc/postgresql"
+
+# If this script is being run by Travis, then we need to point cargo to the libs we installed ourselves.
+# Failure to do this means that the bldr-* components won't be able to find the shared libraries they need.
+# We also need to specify that we want the "travis" user to run things instead of "hab"
+if [ -n "$TRAVIS" ]; then
+  rp=$(find "$HOME/pkgs" -name "lib" -type d | paste -sd ":" -)
+  export LD_RUN_PATH="$LD_RUN_PATH:$rp"
+  user="travis"
+else
+  user="hab"
+fi
+
+if ! exists curl; then
+  echo "curl is required to run the integration tests. Please ensure it's installed and try again."
+  exit 1
+fi
+
+if ! exists hab; then
+  echo "Installing hab"
+  curl https://raw.githubusercontent.com/habitat-sh/habitat/master/components/hab/install.sh | sudo bash
 fi
 
-sudo rm -rf /hab/svc/postgresql
-pushd ../../
+if exists md5sum; then
+  md5=md5sum
+elif exists md5; then
+  md5=md5
+else
+  echo "A program to calculate the md5 hash of a string is required to run the integration tests. Please sort this out and try again."
+  exit 1
+fi
+
+# Install nvm if we don't have it already
+if [ ! nvm &> /dev/null ]; then
+  echo "Installing nvm"
+  curl -o- "https://raw.githubusercontent.com/creationix/nvm/v0.33.8/install.sh" | bash
+fi
+
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh"  ] && . "$NVM_DIR/nvm.sh"
+
+# First make sure that we have services already compiled to test.
+cd $base_dir
+
 make build-srv || exit $?
-env HAB_FUNC_TEST=1 ./support/linux/bin/forego start -f support/Procfile -e support/bldr.env 2>&1 > ./test/builder-api/services.log &
+cd $tmp_dir
+
+if [[ $(uname -a) == *"Darwin"* ]]; then
+  platform="mac"
+else
+  platform="linux"
+fi
+
+name=$(date | $md5 | awk '{ print $1 }')
+dir="$tmp_dir/$name"
+key_dir="$dir/key-dir"
+
+echo "Created $dir for this test run"
+
+mkdir -p $dir $key_dir
+chmod -R 777 $dir $key_dir
+
+env HAB_CACHE_KEY_PATH=$key_dir hab user key generate bldr
+
+if [ -f "$tmp_dir/builder-github-app.pem" ]; then
+  cp "$tmp_dir/builder-github-app.pem" $key_dir
+else
+  cp /root/key-dir/builder-github-app.pem $key_dir
```

Comment:
I think this should be `.secrets/builder-github-app.pem` or `"$base_dir/.secrets/builder-github-app.pem"`

Reply:
This is specific to testing on my VM.  Really, it should be configurable so people can put their keys anywhere.